src/typescript/azure-openapi-validator/tests/individual-azure-tests.ts

```diff
@@ -145,4 +146,27 @@ class IndividualAzureTests {
     const messages: Message[] = await collectTestMessagesFromValidator(fileName, OpenApiTypes.arm, MergeStates.individual)
     assertValidationRuleCount(messages, XmsPageableMustHaveCorrespondingResponse, 0)
   }
+
+  // Failure #1 : RPaaS async response supports 201 only. 202 is not supported.
+  @test public async "Raas Put async operation doesn't support 202"() {
+    const fileName = "RpaasPutAsyncOperationResponseCodeValidation.json"
+    const messages: Message[] = await collectTestMessagesFromValidator(fileName, OpenApiTypes.arm, MergeStates.individual)
+    assertValidationRuleCount(messages, Rpaas_CreateOperationAsyncResponseValidation, 1)
+  }
+
+  // Failure #1 : 'x-ms-long-running-operation' is missing
+  // Failure #2: 'x-ms-long-running-operation-options' is missing
+  @test public async "Raas Put async operation missing x-ms* async extensions"() {
+    const fileName = "RpaasPutAsyncOperationResponseMsCustomExtensionsMissing.json"
+    const messages: Message[] = await collectTestMessagesFromValidator(fileName, OpenApiTypes.arm, MergeStates.individual)
+    assertValidationRuleCount(messages, Rpaas_CreateOperationAsyncResponseValidation, 2)
+  }
+
+  // Failure #1 : 'x-ms-long-running-operation' must be true as operation supports 201 (implies async)
+  // Failure #2: 'final-state-via' must be set to 'azure-async-operation'
+  @test public async "Raas Put async operation is tracked using Auzre-AsyncOperation header"() {
+    const fileName = "RpaasPutAsyncOperationResponseFinalStateViaAzureAsync.json"
+    const messages: Message[] = await collectTestMessagesFromValidator(fileName, OpenApiTypes.arm, MergeStates.individual)
+    assertValidationRuleCount(messages, Rpaas_CreateOperationAsyncResponseValidation, 2)
```

Comment:
This function doesn't compare the message because existing rules always return same message. In case of multiple different messages returned for rpaas rule, it needs to parse messages then compare.